students/Dustin_L/lesson08/sparse_array.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+"""Sparse Array Module"""
+
+
+class SparseArray():
+    """Sparse Array class"""
+    def __init__(self, seq):
+        self.num_elems = 0
+        self.elements = {}
+
+        self.contains_zero = True if 0 in seq else False
+        self.elements = {i: arg for i, arg in enumerate(seq) if arg}
```

Comment:
since the main point of sparse array is not store zeros for efficiency, your dictionary should exclude zero values

Reply:
Hi Natasha,
Perhaps I am missing something, but I am quite sure that my implementation does not store zeros in the self.elements. The conditional "if arg" in the dictionary comprehension should prevent any zeros from being entered as a value in the self.elements dictionary.
The "contains_zero" attribute is primarily for use in the "__contains__" magic method.

I'm pretty sure this works, please let me know if I am missing something.

Reply:
my apologies!! I totally missed the `if arg` part! I will take another look.